File: .github/benchmarks/mach/plugins/transformer.mjs
import { Transformer } from '@alshdavid/mach'

export default new Transformer({
  async transform({ asset, config }) {
    console.log(asset)
    console.log({
      file_path: asset.file_path,
      kind: asset.kind,
      message: "from js transformer", 
      code_length: (await asset.get_code()).length,
    })
  }
})

File: packages/mach/src/plugins/transformer_json/transformer.rs
use crate::public::MachConfig;
use crate::public::MutableAsset;
use crate::public::Transformer;

#[derive(Debug)]
pub struct TransformerJson {}

impl Transformer for TransformerJson {
  fn transform(
    &self,
    asset: &mut MutableAsset,
    _config: &MachConfig,
  ) -> anyhow::Result<()> {
    let code = asset.get_code();

    asset.set_code(&format!("export default ({})", code));
    *asset.kind = "js".to_string();

    return Ok(());
  }
}
